### PR TITLE
Make sure cursor is present in all the callbacks

### PR DESCRIPTION
**Pull request by kamalaknn**

### CODE REVIEW

File: lib/runner/extensions/http-request.command.js [hunk 1]
```diff
@@ -144,7 +144,11 @@ module.exports = {
                     // Since we encountered an error before even attempting to send the request, we bubble it up
                     // here.
                     afterRequest(err, undefined, item.request);
-                    return safeNext(err, {request: item.request}, err);
+                    return safeNext(
+                        err,
```
Comment (kunagpal) [inline lib/runner/extensions/http-request.command.js]: Why is `err` being passed twice?

File: test/integration/request-flow/intermediate-requests.test.js [hunk 2]
```diff
@@ -76,9 +76,11 @@ describe('intermediate requests from auth', function () {
 
         it('must have sent the original request', function () {
             var err = testrun.response.firstCall.args[0],
+                cursor = testrun.response.firstCall.args[1],
                 request = testrun.response.firstCall.args[3];
 
             expect(err).to.be(null);
+            expect(cursor).to.not.be(null);
```
Comment (kunagpal) [inline test/integration/request-flow/intermediate-requests.test.js]: Would it be possible to write a stricter assertion?

Comment (kamalaknn) [reply test/integration/request-flow/intermediate-requests.test.js]: A stricter assertion would be to check the shape of cursor like `expect(testrun.response.firstCall.args[1]).to.have.property('ref');`. Asserting on values is not feasible.

File: lib/runner/extensions/http-request.command.js (hunk 1 above)
Comment (kamalaknn) [reply lib/runner/extensions/http-request.command.js]: Good catch, we had to pass `err` twice to `next`. Now that is handled by `safeNext`.

**Pull request merged**